src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,124 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+// class Ui - handles all the lines to be printed and all the user inputs (commands)
+public class Ui {
+
+    public Ui() {
+
+    }
+
+    // prints out welcome greeting to user when Duke is run
+    public void showWelcome() {
+        System.out.println("    Hi! I'm Samantha\n    How can I help?");
+    }
+
+    public void takeCommands(TaskList t) {
+        Scanner sc = new Scanner(System.in);
+        String s = sc.nextLine();
+
+        ArrayList<Task> tasks = t.getTasks();
+        int taskCounter = tasks.size();
+        while (!s.equals("bye")) {
+            try {
+                // method handleCommand handles current command, returns updated taskCounter variable
+                taskCounter = handleCommand(s, tasks, taskCounter);
+            } catch (Exception e) {
+                System.out.println(e.getMessage());
+            }
+
+            // take in next command
+            s = sc.nextLine();
+        }
+
+        System.out.println("    Bye. Hope to see you soon!");
+    }
+
+    public static int handleCommand(String s, ArrayList<Task> tasks, int taskCounter) {
+        // user enters list command
+        if (s.contains("list")) {
+            if (tasks.isEmpty()) {
+                System.out.println("    No tasks");
+            } else {
+                for (int i = 0; i < taskCounter; i++) {
+                    Task task = tasks.get(i);
+                    System.out.println("    " + (i + 1) + ". " + task);
+                }
+            }
+
+            // user enters mark or unmark command
+        } else if (s.contains("mark") || s.contains("unmark")) {
```

Comment:
This might fail if the user types in an invalid command such as `denmark`. Might also fail for the command `unmark`.

> `denmark` 1

Consider extracting the command and comparing string equality using `String.equals()`.